rendercv/cli/watcher.py

```diff
@@ -0,0 +1,76 @@
+"""
+The `rendercv.cli.watcher` module contains all the functions and classes that are used to watch files and emit callbacks.
+"""
+
+import os
+import pathlib
+import time
+from hashlib import sha256
+from typing import Callable
+
+from watchdog.events import FileModifiedEvent, FileSystemEventHandler
+from watchdog.observers import Observer
+from typer import Exit
+
+
+class ModifiedCVEventHandler(FileSystemEventHandler):
+    """This class handles the file changes and triggers a specified `callback` ignoring duplicate changes.
+
+    Args:
+        file_path (pathlib.Path): The path of the file to watch for.
+        callback (Callable[..., None]): The function to be called on file modification. *CALLBACK MUST BE NON-BLOCKING*
+    """
+
+    file_path: pathlib.Path
+    callback: Callable[..., None]
+    previous_hash: str = ""
+
+    def __init__(self, file_path: pathlib.Path, callback: Callable[..., None]):
+        self.callback = callback
+        self.file_path = file_path
+
+        # Handle an initial pass manually
+        self.on_modified(FileModifiedEvent(src_path=str(self.file_path)))
+
+    def on_modified(self, event: FileModifiedEvent) -> None:
+        if event.src_path != str(self.file_path):
+            # Ignore any events that aren't our file.
+            return
+
+        file_hash = sha256(open(event.src_path).read().encode("utf-8")).hexdigest()
+
+        if file_hash == self.previous_hash:
```

Comment:
If `on_modified` is only called when a file has been modified, why are we still checking if it has been changed?

Reply:
The on_modified event can trigger regardless of the contents. My use case was that these events occur on each save of a file in VS Code but the same problem occurs in other editors like Vim.

Reply:
Is there documentation somewhere that lists the events that trigger `on_modified` exactly? It makes sense to me to be triggered when a file is saved; it implies it has changed, right?

Reply:
The documentation for the [on_modified](https://pythonhosted.org/watchdog/api.html#watchdog.events.FileSystemEventHandler.on_modified) event says.

> Called when a file or directory is modified.

If you want I can make a decorator to split the logic out.

```py
@cached__from_last_file
def on_modified(self, event: FileModifiedEvent) -> None:
    ...
 ```
 
This also brought to my attention that I did not include a test case for duplicate values, so I will be including that aswell.